lib/matplotlib/ticker.py

```diff
@@ -1799,6 +1806,14 @@ def __call__(self):
         return self.tick_values(vmin, vmax)
 
     def tick_values(self, vmin, vmax):
+        if self.numticks == 'auto':
+            if self.axis is not None:
+                numticks = max(min(self.axis.get_tick_space(), 9), 2)
```

Comment:
I understand the minimum of 2, but 9 seems a bit of a random number. Why was this chosen?